samples/performance/multithreading_render_passes/multithreading_render_passes_tutorial.md

```diff
@@ -0,0 +1,99 @@
+
+ The diagram below shows this two step process:
+
+![Render Passes Diagram](images/render_passes_diagram.png)
+
+ Note that there is a dependency because the second pass is using the output of the first. Since these are two separate render passes we cannot use a ``VkSubpassDependency`` for synchronization. Instead ``VkImageMemoryBarrier`` is used.
+
+## The Multi-threading Render Passes Sample
+
+If we have two or more render passes we can record them separately in different threads. 
+
+Note that the more similar is the workload for all the passes, the more performance improvement you can get by splitting the work between multiple threads. In this sample the same scene is rendered once in each render pass but from different viewpoints and with different complexity of commands recording (shadow pass requires less descriptor and resources setup for each frame). That means the workloads are not fully equal, but it is still worth delegating part of the work to another thread, and the increase in performance is noticeable.
+
+One way to use multi-threading with multiple render passes is to create a separate primary level command buffer for each of them. In this case command buffers can be recorded independently and then submitted to the queue all at once using ``vkQueueSubmit``.
+
+Another approach is to use secondary level command buffers. First, both of the passes are recorded into two separate secondary command buffers using two threads. Then, we can just reference them in the primary command buffer via ``vkCmdExecuteCommands``.
+
+When using both of these methods for multi-threading, general recommendations should still be taken into account (see [Multi-threaded-recording](https://github.com/KhronosGroup/Vulkan-Samples/blob/master/samples/performance/command_buffer_usage/command_buffer_usage_tutorial.md#Multi-threaded-recording)).
+
+This sample shows the difference between recording both render passes into a single command buffer in one thread and using the methods described above.
+
+Below are screenshots of the sample running on a phone with a Mali G72 GPU:
+
+![Single Thread](images/no_multi_threading.png)
+
+Using two threads gives a 10ms frame time improvement and CPU cycles show an increase in CPU utilization:
+
+![Primary Command Buffers](images/primary_command_buffers.png)
+
+With secondary command buffers you can see a further drop in frame time.
+
+![Secondary Command Buffers](images/secondary_command_buffers.png)
+
+## Profiling
+
+Android Profiler can help to see how threads are utilized. Flame Chart shows how much time was spent for each function execution during a particular timeframe. In this particular example total contribution of command buffers recording in the main thread is 9.94 seconds within a 10 seconds capture with multi-threading disabled. 
+
```

Comment:
Can we make this a bit more generic and less android specific? i.e. "A profiling tool, such as Android Profiler, can help..."